src/main/java/net/snowflake/client/core/HttpClientSettingsKey.java

```diff
@@ -15,35 +15,34 @@
 public class HttpClientSettingsKey implements Serializable {
 
   private OCSPMode ocspMode;
-  private boolean useProxy;
+  private boolean useProxy = false;
   private String proxyHost = "";
   private int proxyPort = 0;
   private String nonProxyHosts = "";
   private String proxyUser = "";
   private String proxyPassword = "";
   private String proxyProtocol = "http";
+  private String userAgentSuffix = "";
 
   public HttpClientSettingsKey(
       OCSPMode mode,
+      boolean useProxy,
       String host,
       int port,
       String nonProxyHosts,
       String user,
       String password,
-      String scheme) {
-    this.useProxy = true;
```

Comment:
Default was true. So if we add new constructor semantic will remain same.

Reply:
Default depends implicitly on the usage of a certain constructor which provides poor encapsulation as the API user has to inspect the constructor code to understand the value of userProxy

Reply:
Reverted to existing semantics by getting the old constructors back.